### PR TITLE
Add coloured team-doors

### LINKED ISSUE #149: Team doors control
How exactly should work team doors?
Yesterday I picked up some doors and placed them somewhere in the center of the map.
Immediately after this, I tried to open them, but they stay closed. After some time I have seen that they were opened by member of enemy team.

I expect that team doors placed by member of team A should always be available to members of team A.

### PR DESCRIPTION
**Pull request by ClobberXD**
Closes #149 
### Proposed changes (as suggested in the aforementioned issue)
- Add coloured team-doors for visual distinction.
- Players can only have normal steel doors, which convert into their respective team-doors on place.
- Team-doors can be opened by anyone from the placer's team, _irrespective of its location in the map_ (which is the expected behaviour).

### CODE REVIEW

**Comment by rubenwardy** [comment]
you shouldn't be able to place them on the other side

**Comment by ClobberXD** [comment]
Why not?

**Comment by rubenwardy** [comment]
> door-placement could be prohibited within an n-node radius of the enemy base.

Please may you do this

**Comment by ClobberXD** [comment]
Assuming n == 25

**Comment by rubenwardy** [comment]
Probably, but that can be done later if you like

**Comment by ClobberXD** [comment]
Added proximity check. Uses hard-coded team names though, which isn't the best solution. Should this be modified to iterate through all teams and check for proximity with each team base which isn't the placer's?

File: mods/doors/init.lua [hunk 1]
```diff
@@ -302,6 +302,15 @@ function doors.register(name, def)
 			-- Get placer's team
 			local tname = ctf.player(pn).team or ""
 
+			-- Prevent door placement if within 40 nodes of enemy base
+			local enemy_base, enemy_team
+			enemy_team = tname == "red" and "blue" or "red"
```
Comment (rubenwardy) [inline mods/doors/init.lua]: please move local to the same line as init, ie:

```lua
local variable = -- ...
```

File: mods/doors/init.lua [hunk 2]
```diff
@@ -380,6 +402,13 @@ function doors.register(name, def)
 	def.after_dig_node = function(pos, node, meta, digger)
 		minetest.remove_node({x = pos.x, y = pos.y + 1, z = pos.z})
 		minetest.check_for_falling({x = pos.x, y = pos.y + 1, z = pos.z})
+
+		-- After dig, convert coloured doors into normal steel doors
+		local wielditem = digger:get_wielded_item()
```
Comment (rubenwardy) [inline mods/doors/init.lua]: this doesn't work, use item drops instead

Comment (ClobberXD) [reply mods/doors/init.lua]: Good idea. Will do

**Comment by ClobberXD** [comment]
Doors used to go to a separate slot on dig, now I see why. Switched to using node drops.

File: mods/doors/init.lua (hunk 1 above)
Comment (ClobberXD) [reply mods/doors/init.lua]: Done

**Pull request merged**